packages/cascara/src/ui/Table/TechnicalDesign.mdx

```diff
@@ -0,0 +1,138 @@
+
+### Config
+
+There will be use cases where features like multi-selection, CRUD, custom actions, etc. won't be required at all, they must be disabled by thefault but easily activated. It makes sense to have boolean props that controll such features.
+
+Another important aspect of configuration is the definition of columns, hence a complete section is devoted to it.
+
+| Name    | Type          | default | Description                                        |
+| ------- | ------------- | ------- | -------------------------------------------------- |
+| Bulk    | Boolean       | `FALSE` | If true, activates the bulk mode (multi-select)    |
+| Columns | Array[Object] | `[]`    | Holds the definitions for each column in the table |
+| Create  | Boolean       | `FALSE` | If true, activates the creation of new items       |
+| Delete  | Boolean       | `FALSE` | If true, activates the deletion of items           |
+| Update  | Boolean       | `FALSE` | If true, activates the edit mode                   |
+
+### Triggers
+
+Custom actions are present in most tables, they are an important aspect of all applications. The nature of the actions depends on the entity they are applied to, hence, an abstract way of defining custom actions is required.
+
+| Attribute | Type     | default      | Description                                                 |
+| --------- | -------- | ------------ | ----------------------------------------------------------- |
+| id        | String   | `''`         | The id of the action, usually the name.                     |
+| callback  | Function | `noop`       | The function to be called when the action button is clicked |
+| trigger   | ANY      | `<Button />` | The element that will trigger the action                    |
+
+### Data
+
+The data passed to the table for rendering purposes.
+
+| Name | Type          | default | Description                                   |
+| ---- | ------------- | ------- | --------------------------------------------- |
+| data | Array[Object] | `[]`    | the tabular data to be displayed by the table |
+
+### Columns
+
+The definition of columns depends on the data, we must make sure we have a well-defined set of data types to allow display and mutation for each one.
+
+| Attribute  | Type                                                                   | default    | Description                                              |
+| ---------- | ---------------------------------------------------------------------- | ---------- | -------------------------------------------------------- |
+| accessor   | String                                                                 | `null`     | The name of the entity attribute that maps to the column |
+| context    | Object                                                                 | `{}`       | the context to be passed down to the custom component    |
```

Comment:
Walk me through the use case with this. Is this actually React context?

Reply:
No, this was intended to be dot-spreaded as props in the rendered context.

I was thinking on the `IntentLookup` component that expects the Redux mapped action to be passed, my plan was to dot-spread it on the component passed as `render`.

Reply:
@lordpixel Okay, thanks for making this more clear.

1. We want to stay away from anything internal to these components relying on external implementation at all. That means these should not ever need to rely on something like immutable data libraries or state actions from something like Redux. This is because if those external dependencies change or get removed, our components internally could break. These need to work regardless of how that data is getting picked up.
2. Knowing the above... and assuming this ends up being for something like `dataConfig` that is reusable, I wonder if there is a pattern we could use to compose those functions outside of the component where they would be ready to handle this logic as a handler passed into the component. We could have a pretty complex handler function that is supposed to do whatever. The Redux action could be inside that handler function outside of the table. Maybe the onClick function for an action would also return the data relative to where that action was clicked. That way it could be consumed externally in the handler function. I guess in the case of actions, that would mean the onClick would also return the data for the row? 

It's good you put this parameter here. I can now see there are scenarios we would need to know current state of the data at the time of an action firing. And that could only come from inside the component. On this note... I think SUIR is actually doing something similar to this by adding data about the component to React's synthetic events for onClick, etc. Let me go look for some examples of that real quick. It might be the best way to make this more simple. Expose less of the logic internal and still keep a simple external API.

Reply:
Yeah, @lordpixel  take a look at this: https://github.com/Semantic-Org/Semantic-UI-React/blob/master/src/elements/Button/Button.js#L64 

Maybe we do something similar, but for each of those actions, with onClick on the row, we could return the current data for the row itself, not for the button. This could get tricky when dealing with a row being edited. I guess actually that is okay. It should probably return the actual data for the row, even if it is not being shown based on dataConfig. Just return the object for the row from `data`. If it is a form being edited, the logic for doing that will be contained in the Form itself.

Reply:
re. #1. and #2.: Not wanting to rely on Redux, just wanted to point out that it was the logic coming from outside, Redux was just an example. This design is completely stack-agnostic.

re. SUIR implementation of events: I've always liked the way an input passes data pretaining to the component itself as well as its current value, we should indeed replicate that pattern, I will keep this in mind when trhowing some code, Thanks!

Reply:
I see. So not necessarily doing anything with that Redux action itself, but providing it to the action so that we know what the related function is when the action is taken. Another option... is that you could just map on data before passing to the table and add those items to each object in the row if you knew what they were before passing to the data prop. That would work great if we then return the data for that row with each of the action onClick events. You would get back the data for that particular row... which would include the Redux action you passed in to that object when creating the data. If that was the intent... we would need to put a decent example of that implementation pattern together for someone to easily understand that.

```
const newData = originalData.map(data) => data.actions = actionsObjectWithFuncOrWhatever

<Table data={newData} dataConfig={...} />
```

This would essentially eliminate the need for this context prop.